File: app/src/main/java/com/nimtego/plectrum/domain/interactor/ArtistInteractor.java
package com.nimtego.plectrum.domain.interactor;

import com.nimtego.plectrum.presentation.main.model.ArtistModel;

import java.util.List;

import dagger.internal.Preconditions;
import io.reactivex.Observable;

@Deprecated
public class ArtistInteractor extends BaseInteractor<List<ArtistModel>, ArtistInteractor.Params> {

    public ArtistInteractor() {
    }

    @Override
    protected Observable<List<ArtistModel>> buildUseCaseObservable(Params params) {
        Preconditions.checkNotNull(params);
        return repository.artists(params.request);
    }

    public static final class Params {

        private final String request;

        private Params(String request) {
            this.request = request;
        }

        public static Params forRequest(String request) {
            return new Params(request);
        }
    }

}

File: app/src/main/java/com/nimtego/plectrum/domain/interactor/ArtistInteractor.kt
package com.nimtego.plectrum.domain.interactor

import com.nimtego.plectrum.presentation.main.model.ArtistModel
import io.reactivex.Observable


class ArtistInteractor : BaseInteractor<List<ArtistModel>, ArtistInteractor.Params>() {

    override fun buildUseCaseObservable(params: Params): Observable<List<ArtistModel>> {
//        Preconditions.checkNotNull(params)
        return repository.artists(params.request)
    }

    class Params private constructor(val request: String) {
        companion object {

            fun forRequest(request: String): Params {
                return Params(request)
            }
        }
    }

}